sw/device/tests/spi_passthru_test.c

```diff
@@ -0,0 +1,96 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/json/spi_passthru.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_spi_device.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/json/command.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_flow_control.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
+
+static dif_spi_device_handle_t spid;
+
+status_t configure_jedec_id(ujson_t *uj, dif_spi_device_handle_t *spid) {
+  config_jedec_id_t config;
+  TRY(ujson_deserialize_config_jedec_id_t(uj, &config));
+  dif_spi_device_flash_id_t id = {
```

Comment:
Just a thought: The logic for setting up the command and IDs could probably go into spi_device_testutils at some point.

Reply:
I think we can make a judgement about how to handle that in the follow-on PRs.